File: fibonacci.py

def fibonacci (num):
    fib = [0, 1]
    for i in range(2, num+1):
        fib[i%2] = fib[0] + fib[1]

    if (num%2)==0 :
        return fib[0]
    else:
        return fib[1]


def main():
    print( fibonacci(7) )



if __name__ == "__main__": main()

File: fibonacci.cpp
#include "stdio.h"

// iterative
int fibonacci (int in) {
  int fib[2] = {0, 1};

  for (int i=2; i<=in; ++i) {
    fib[i%2] = fib[0] + fib[1];
        //printf("%d\n", fib[i%2]);      
  }
    
  if(in%2==0)
    return fib[0];
  else
    return fib[1];
}

int main() {
  printf ("%d\n", fibonacci(7));
}
